client/apiversion_test.go

```diff
@@ -0,0 +1,99 @@
+package client_test
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+type versionHandler struct {
+	authBody string
+	port     string
+}
+
+type MakeServiceURLTest struct {
+	serviceType string
+	version     string
+	parts       []string
+	success     bool
+	URL         string
+	err         string
+}
+
+func (s *localLiveSuite) makeServiceURLTests() []MakeServiceURLTest {
+	return []MakeServiceURLTest{
+		{
+			serviceType: "compute",
```

Comment:
Given object-store is special, there should be MakeServiceURL tests for URLs for things other than just compute?